File: solutions/julia/pangram/1/pangram.jl
"""
    ispangram(input)

Return `true` if `input` contains every alphabetic character (case insensitive).

"""
function ispangram(input)
    letters=Dict()
    for chr in input
        if chr >= 'a' && chr <= 'z' chr=chr-32 end
        if chr >= 'A' && chr <= 'Z' letters[chr]=true end
    end
    length(letters)==26
end

File: solutions/haskell/pangram/1/src/Pangram.hs
module Pangram (isPangram) where

isPangram :: String -> Bool
isPangram "" = False
isPangram text = all exists ['a'..'z']
  where
    exists chr = contains chr text || contains ((toEnum ((fromEnum chr) - 32)) :: Char) text
    contains chr [] = False
    contains chr (x:xs) = if chr == x then True else contains chr xs
